python/flight-crew-scheduling/src/flight_crew_scheduling/domain.py

```diff
@@ -0,0 +1,177 @@
+        return skill in self.skills
+
+    def is_available(self, from_date_inclusive: date, to_date_inclusive: date) -> bool:
+        """Checks if the employee is available between two dates."""
+        if len(self.unavailable_days) == 0:
+            return True
+        current_date = from_date_inclusive
+        while current_date <= to_date_inclusive:
+            if current_date in self.unavailable_days:
+                return False
+            current_date += timedelta(days=1)
+        return True
+
+    def __str__(self) -> str:
+        return self.name
+
+    def __repr__(self) -> str:
+        return (f"Employee(id={self.id}, name={self.name}, home_airport={self.home_airport}, skills={self.skills},"
+                f" unavailable_days={self.unavailable_days})")
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Employee):
+            return False
+        return self.id == other.id
+
+    def __hash__(self) -> int:
+        return hash(self.id)
+
+
+class Flight(JsonDomainBase):
+    flight_number: Annotated[str, PlanningId]
+    departure_airport: Annotated[Airport | None, IdSerializer, AirportDeserializer, Field(default=None)]
+    departure_utc_date_time: Annotated[datetime | None, Field(default=None, alias="departureUTCDateTime")]
+    arrival_airport: Annotated[Airport | None, IdSerializer, AirportDeserializer, Field(default=None)]
+    arrival_utc_date_time: Annotated[datetime | None, Field(default=None, alias="arrivalUTCDateTime")]
+
+
+    def get_departure_utc_date(self) -> date:
+        """Retrieve flight's departure date."""
+        return self.departure_utc_date_time.date()
+
+    def __str__(self) -> str:
+        return f"{self.flight_number}@{self.get_departure_utc_date()}"
+
+    def __repr__(self) -> str:
+        return (f"Flight(flight_number={self.flight_number}, departure_airport={self.departure_airport}, departure_utc_date_time={self.departure_utc_date_time},"
+                f" arrival_airport={self.arrival_airport}, arrival_utc_date_time{self.arrival_utc_date_time})")
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Flight):
+            return False
+        return self.flight_number == other.flight_number
+
+    def __hash__(self) -> int:
+        return hash(self.flight_number)
+
+    def __lt__(self, other: "Flight") -> bool:
+        if self.departure_utc_date_time != other.departure_utc_date_time:
+            return self.departure_utc_date_time < other.departure_utc_date_time
+        if self.departure_airport != other.departure_airport:
+            return self.departure_airport < other.departure_airport
+        if self.arrival_utc_date_time != other.arrival_utc_date_time:
+            return self.arrival_utc_date_time < other.arrival_utc_date_time
+        if self.arrival_airport != other.arrival_airport:
+            return self.arrival_airport < other.arrival_airport
+        return self.flight_number < other.flight_number
```

Comment:
Tuple comparison logic can simplify this:

```suggestion
    def _comparison_key(self):
        return (departure_utc_date_time, departure_airport, arrival_utc_date_time, arrival_airport, flight_number)
    
    def __lt__(self, other: "Flight") -> bool:
        return self._comparison_key() < other._comparison_key()
```